pkg/controller/controller_binding.go

```diff
@@ -824,34 +824,6 @@ func (c *controller) serviceBindingRequestUnbinding(binding *v1beta1.ServiceBind
 
 	response, err := brokerClient.Unbind(unbindRequest)
 	if err != nil {
-		if httpErr, ok := osb.IsHTTPError(err); ok {
```

Comment:
That is my understanding. As it is written in master, if there is a dial error, then service catalog does not attempt to resend the unbind request. Instead, service catalog should be attempting to resend the unbind request for the entire reconciliation duration regardless of whether the error is a dial error or a broker error. There is no meaningful differentiation for unbind requests.

Reply:
If you look at what is being done for deprovision, the only difference between a dial error and a broker error is the format of the message that service catalog outputs to the log et al.

Reply:
In master today, there is no differentiation between the Ready condition message for an HTTP status code error response and the unbind call attempt having an error. They both say `Unbind call failed. Error unbinding from X: $error-text`. Since there was no difference in that message, I could just remove the special case here entirely, as the only thing unique here was the handling of marking the resource as failed.

If we'd like to differentiate the message like deprovision does, that's fine as well, but I was just keeping consistency with the behavior of unbinds today.

Reply:
I think it's fine the way it is for now - you're right that it's just a matter of the message we put on it when it comes to unbinds.  It matters a lot more for provision and bind since those affect whether orphan mitigation is done.